modify search to use inlineable sql

bitner opened the pull request:
Modifies search functions to use straight SQL functions which allow the query planner to inline the sql which should perform quite a bit better than with PLPGSQL.

sharkinsspatial commented:
@bitner This threw a syntax error when running `docker-compose up -d`.  Were you able to run this locally and run the integration tests with `yarn test_rest`?

bitner commented:
fixed some typos, rest tests pass, but the pgTap tests need to be updated still

bitner commented:
I fixed the database tests. check_test is really only used in pg_tap when you want to interactively test a test just running sql directly. When calling with the test runner, you want to call the tests functions directly. Added missing view 'root' in the views test. Added plan counts which gives better output with the test runner.
